feat: Pendle V2

SinSkylark opened the pull request:
# Pendle V2 Integration

## Description
Comprehensive implementation of Pendle V2 integration with complete functionality:

### Core Features
- ✅ Market Core Actions
  - Add/Remove Liquidity (Single/Dual token operations)
  - Market Oracle interactions
  - Standardized Yield operations
- ✅ Router Functions
  - Initialize router
  - ETH liquidity management
  - Token swaps
- ✅ Limit Orders
  - Fill orders
  - Cancel orders (single/batch)
  - Order status tracking
- ✅ Standardized Yield
  - Deposit/Redeem operations
  - Exchange rate queries
  - Token validation
  - Reward management
- ✅ Advanced Features
  - Market Oracle observations
  - Preview functions for deposits/redemptions
  - Gas-optimized operations

### Technical Improvements
- Enhanced error handling with detailed messages
- Comprehensive input validation
- Type-safe implementations
- Improved response formatting
- Complete mock provider with all handlers
- Extensive test coverage

## Supported Networks
- Ethereum (Mainnet)
- Arbitrum One
- Sepolia (Testnet)

## Test Results
- All test suites passing (8/8):
  - ✅ Market Core Actions
  - ✅ Router Functions
  - ✅ Market Oracle
  - ✅ Limit Orders
  - ✅ Add/Remove Liquidity
  - ✅ Standardized Yield
  - ✅ Redeem Rewards
  - ✅ Swap Functions

## Implementation Details
- Modular architecture with separate function modules
- Comprehensive mock provider for testing
- Robust error handling and validation
- Full TypeScript type safety
- Follows project coding standards
- Gas-optimized operations

## Features
- Market information retrieval
- Liquidity management (Single/Dual token)
- Token swaps and routing
- Limit order operations
- Standardized yield operations
- Reward claiming and tracking
- Market oracle interactions
- Preview functions for operations

## Next Steps
- Add documentation for advanced features
- Implement additional market validations
- Add performance optimizations
- Expand cross-chain functionality
- Add more example use cases

## References
- [Pendle Finance](https://www.pendle.finance/)
- Protocol Type: Yield Trading & Management
- Twitter: [@pendle_fi](https://twitter.com/pendle_fi)
- [Documentation](https://docs.pendle.finance/)

SinSkylark committed:
feat(pendle-v2): Initial implementation of getMarketInfo and claimRewards functions with passing tests

SinSkylark committed:
Refactor: Clean up and ensure all files are contained within the projects directory

This commit includes restructuring the pendle-v2 project to ensure all files are properly organized within the projects directory. Fixed TypeScript errors and verified that all tests pass successfully.

SinSkylark committed:
feat: implement liquidity management functions - Add comprehensive liquidity management functionality including add/remove liquidity functions with dual token, single token, and SY support. Add corresponding ABIs, interfaces, mock provider responses, and test coverage.

SinSkylark committed:
feat: implement core Pendle v2 functions - Market Core (addLiquidity/removeLiquidity, redeemRewards, isExpired), Router Operations (initialize, ETH pairs, token swaps), Cross-chain Messaging (fee calc, message handling), Merkle Distribution (claims, verification, tracking)

SinSkylark committed:
refactor: remove deprecated ABIs and enhance address validation in liquidity functions

- Deleted unused ABIs from `abis/index.ts` to streamline the codebase.
- Updated `addLiquidity` and `removeLiquidity` functions to improve address validation, returning detailed error messages for invalid addresses.
- Refactored validation logic in `validation.ts` to return boolean values instead of throwing errors, enhancing error handling in the application.

SinSkylark committed:
feat: enhance Pendle V2 integration with comprehensive functionality - Add standardized yield functions, market core actions, router functions, limit orders, improved error handling, and complete test suite

OoXooOx commented on projects/pendle-v2/functions/actions/actionAddRemoveLiq.ts:
```diff
@@ -0,0 +1,341 @@
+        validateAddress(market);
+        validateAddress(input.tokenIn);
+        validateAddress(input.tokenMintSy);
+
```
getProvider(); accept chainId as argument

SinSkylark replied:
Thank you for your review. I completely missed it. I'll turn on and check notifications for this PR. I have resolved the comment.

Should I continue updating other action files to include chainId support?

SinSkylark committed:
feat: add chainId support for provider and liquidity functions

- Updated `getProvider` to accept an optional chainId parameter
- Modified `addLiquidityDualTokenAndPt` and `addLiquiditySingleToken` to pass chainId to provider
- Updated type definitions to reflect new provider signature
- Added logging for provider chain selection
